examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImBatchInvokeCommand.kt

```diff
@@ -0,0 +1,182 @@
+      setFailure("invoke failure")
+    }
+
+    override fun onResponse(invokeResponseData: InvokeResponseData) {
+      logger.log(Level.INFO, "Batch Invoke receive OnResponse on $invokeResponseData")
+      val clusterId = invokeResponseData.getClusterId().getId()
+      val commandId = invokeResponseData.getCommandId().getId()
+      val tlvData = invokeResponseData.getTlvByteArray()
+      val jsonData = invokeResponseData.getJsonString()
+      val status = invokeResponseData.getStatus()
+
+      if (clusterId == CLUSTER_ID_IDENTIFY && commandId == IDENTIFY_COMMAND) {
+        if (tlvData != null || jsonData != null) {
+          setFailure("invoke failure with problematic payload")
+        }
+        if (
+          status != null && status.status != Status.Code.Success && status.clusterStatus.isPresent()
+        ) {
+          setFailure("invoke failure with incorrect status")
+        }
+      }
+
+      if (clusterId == CLUSTER_ID_TEST && commandId == TEST_ADD_ARGUMENT_RSP_COMMAND) {
+        if (tlvData == null || jsonData == null) {
+          setFailure("invoke failure with problematic payload")
+        }
+
+        if (!jsonData.equals("""{"0:UINT":2}""")) {
+          setFailure("invoke failure with problematic json")
+        }
+
+        if (status != null) {
+          setFailure("invoke failure with incorrect status")
+        }
+      }
+      responseCount++
+    }
+
+    override fun onNoResponse(noInvokeResponseData: NoInvokeResponseData) {
+      logger.log(Level.INFO, "Batch Invoke receive onNoResponse on $noInvokeResponseData")
+      responseCount--
+    }
+
+    override fun onDone() {
+      if (responseCount == TEST_COMMONDS_NUM) {
+        setSuccess()
+      } else {
+        setFailure("invoke failure")
+      }
+    }
+  }
+
+  private inner class InternalGetConnectedDeviceCallback : GetConnectedDeviceCallback {
+    override fun onDeviceConnected(devicePointer: Long) {
+      setDevicePointer(devicePointer)
+      logger.log(Level.INFO, "onDeviceConnected")
+    }
+
+    override fun onConnectionFailure(nodeId: Long, error: Exception) {
+      logger.log(Level.INFO, "onConnectionFailure")
+    }
+  }
+
+  override fun runCommand() {
+    val number: UShort = 1u
+    val tlvWriter1 = TlvWriter()
+    tlvWriter1.startStructure(AnonymousTag)
+    tlvWriter1.put(ContextSpecificTag(0), number)
+    tlvWriter1.endStructure()
+
+    val element1: InvokeElement =
+      InvokeElement.newInstance(
+        /* endpointId= */ 0,
+        CLUSTER_ID_IDENTIFY,
+        IDENTIFY_COMMAND,
+        tlvWriter1.getEncoded(),
+        null
+      )
+
+    val tlvWriter2 = TlvWriter()
+    tlvWriter2.startStructure(AnonymousTag)
+    tlvWriter2.put(ContextSpecificTag(0), number)
+    tlvWriter2.put(ContextSpecificTag(1), number)
+    tlvWriter2.endStructure()
+
+    val element2: InvokeElement =
+      InvokeElement.newInstance(
+        /* endpointId= */ 1,
+        CLUSTER_ID_TEST,
```

Comment:
Is this test cluster there for all different servers?

I am trying to understand why we are deciding to send this second command here. Is this a temporary things, or is this expected to be more permanent?

Reply:
This is the test, we pair firstly, then send two command in batch, thanks

Reply:
Not permanent, this is temporary, we need the dedicated IM folder, and dedicated test suite that use IM for testing, that effort is still in radar , would happen in two or three months

Reply:
it would be handled in https://github.com/project-chip/connectedhomeip/issues/25649, thanks